app/jobs/send_invitations_job.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class SendInvitationsJob < ApplicationJob
+  queue_as :default
+
+  def perform(invitation_set)
+    any_valid = false
+    invitation_text = ''
+    invitation_set.reload
+    invitation_set.responses.opened_and_not_expired.each do |response|
+      invitation_text = random_message(response)
+      any_valid = true
+      break
+    end
+    return unless any_valid
+    invitation_token = invitation_set.invitation_tokens.create!
+    plain_text_token = invitation_token.token_plain
+    invitation_set.update_attributes!(invitation_text: invitation_text) if invitation_set.invitation_text.blank?
+    send_invitations(invitation_set, plain_text_token)
+  end
+
+  def max_attempts
+    2
+  end
+
+  def reschedule_at(current_time, _attempts)
+    current_time + 1.hour
+  end
+
+  private
+
+  def send_invitations(invitation_set, plain_text_token)
+    invitation_set.invitations.each do |invitation|
+      invitation.sending!
+      SendInvitationJob.perform_later(invitation, plain_text_token)
+    end
+  end
+
+  def random_message(response)
+    if response.protocol_subscription.person.mentor?
+      mentor_texts(response)
+    else # Student
+      response.substitute_variables(StudentInvitationTexts.message(response.protocol_subscription.protocol,
```

Comment:
die `substitute_variables` functie is volgens mij iets dat we uit responses moeten halen. Niet voor nu, maar misschien voor als we ons hele ding gaan roquafyen voor de ortho mensen.

Reply:
Check. Ik maak wel een frietkempje.